src/components/Modals/SetAclPermissionModal.jsx

```diff
@@ -0,0 +1,200 @@
+// React Imports
+import React, { useContext, useState } from 'react';
+// Custom Hook Imports
+import { useSession, useStatusNotification } from '@hooks';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import ClearIcon from '@mui/icons-material/Clear';
+import Dialog from '@mui/material/Dialog';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import SettingsIcon from '@mui/icons-material/Settings';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+// Utility Imports
+import { runNotification, setDocAclPermission } from '@utils';
+// Context Imports
+import { SignedInUserContext } from '@contexts';
+// Component Imports
+import DocumentSelection from '../Form/DocumentSelection';
+import FormSection from '../Form/FormSection';
+
+/**
+ * @typedef {import("../../typedefs.js").setAclPermissionModalProps} setAclPermissionModalProps
+ */
+
+/**
+ * SetAclPermissionModal Component - Modal component that generates the form for
+ * setting ACL permissions to another user's Document in their Solid
+ * Pod via Solid Session
+ *
+ * @memberof Forms
+ * @name SetAclPermissionModal
+ * @param {setAclPermissionModalProps} Props - Props for SetAclPermissionModal component
+ * @returns {React.JSX.Element} The SetAclPermissionModal Component
+ */
+const SetAclPermissionModal = ({ showModal, setShowModal, documentName }) => {
+  const { session } = useSession();
+  const { state, dispatch } = useStatusNotification();
+  const { podUrl } = useContext(SignedInUserContext);
+  const [docType, setDocType] = useState('');
+  const [permissionState, setPermissionState] = useState({
+    podUrlToSetPermissionsTo: '',
+    permissionType: ''
+  });
+
+  const handleDocType = (event) => {
+    setDocType(event.target.value);
+  };
+
+  const clearInputFields = () => {
+    dispatch({ type: 'CLEAR_PROCESSING' });
+  };
+
+  const handleCloseModal = () => {
+    setPermissionState({
```

Comment:
I think we can have `setPermissionState` and `setDocType` be moved to `clearInputFields` rather than a handler function for closing the modal.

Reply:
Was not expecting `setShowModal` to move as well, but guess this works.